src/compontens/Blog.js

```diff
@@ -1,33 +1,86 @@
-import React from 'react';
+import React, {useState} from 'react';
 import { Link } from 'react-router-dom';
 import {AiOutlineLike, AiOutlineDislike} from 'react-icons/ai';
 import {CgComment} from 'react-icons/cg';
+import {getTime} from '../hooks/getTime';
+
+export default function Blog({blog, reactions, children}) {
+  const time = getTime(blog.blog_date);
+
+  const renderedReactions = reactions.filter((reaction)=>{
+    return reaction.blog_id === blog.id;
+  })
+
+  const like = renderedReactions.filter((reaction)=>{
+    return reaction.reaction_type === 'like';
+  })
+
+  const dislike = renderedReactions.filter((reaction)=>{
+    return reaction.reaction_type === 'dislike';
+  })
+
+  const [upvote, setUpvote] = useState(like.length);
+  const [downvote, setDownvote] = useState(dislike.length);
+
+  const handleReactionClick = async(reaction_type, purpose)=>{
+    try {
+
+      const body = {
+        blog_id: blog.id,
+        reaction_type,
+        purpose
+      } 
+      const response = await fetch(`${process.env.REACT_APP_BASE_URL}/blogs/reaction`, {
```

Comment:
See this here, this works but you are constantly going to be writing these headers, try axios. There is something called axios interceptor, it will automate this process for you and you can split auth endpoints and non-auth endpoints really easy.

Reply:
I do axios in some earlyer projects, for me it's easier with fetch, but probably i will switch to axios.

Reply:
earlier...